### PR TITLE
[16.0][IMP] stock_picking_mass_action: Added the possibility to indicate to the method if we want other types of operation.

### PR DESCRIPTION
**Pull request by RodrigoBM**

### CODE REVIEW

**Review by AlhMaria — APPROVED** [review]

**Comment by FranciscoFactorLibre** [comment]
I have checked and it seems correct.

I show applied flow, and tests.

## Configure warehouse

1. Active **Multi-Step Routes**

image

2. Create new warehouse, outgoing configuration (2 Steps)

image



## Sales flow and Stock Pikcing create

1. Create and confirm orders in both warehouses


https://github.com/OCA/stock-logistics-workflow/assets/38328941/ca834cec-e7d1-4d85-9e48-5b595c2c5554

2. Add stock in both warehouses, for products


https://github.com/OCA/stock-logistics-workflow/assets/38328941/a36aaed5-5b69-475b-b9d8-ef6473bf513c

## Check Availability

1. Filter Picking

image

2. Check Availability without context

https://github.com/OCA/stock-logistics-workflow/assets/38328941/86e79d45-c96c-4cb8-bec2-e35d51370fb8

2. Check Availability with context operation type _outgoing_ and _incoming_
   * Context used: `model.check_assign_all(other_domain=[("picking_type_code", "in", ["outgoing", "incoming"])])` 


https://github.com/OCA/stock-logistics-workflow/assets/38328941/fa697292-cb69-45a8-a02e-b95ec71a9a32

2. Check Availability with context operation type _outgoing_ and _internal_
   * Context used: `model.check_assign_all(other_domain=[("picking_type_code", "in", ["outgoing", "internal"])])` 


https://github.com/OCA/stock-logistics-workflow/assets/38328941/93b7293d-b4bd-45fa-8be4-32fe1536f802


2. Check Availability with context operation type _outgoing_ and _internal_ and picking type id `2,15`
    * Context used: `  model.check_assign_all(other_domain=[("picking_type_code", "in", ["outgoing", "internal"]) and ("picking_type_id", "in", [2,15])])` 
    * Context used: `  model.check_assign_all(other_domain=[("picking_type_code", "in", ["outgoing", "internal"]) and ("picking_type_id", "in", [15])])` 

https://github.com/OCA/stock-logistics-workflow/assets/38328941/c605525b-204a-4b1e-b492-266b0ba994b8



-----------

This application of the functionality is very useful to manage the availability check of several warehouses, or with multi-stage warehouses automatically.

**Review by FranciscoFactorLibre — APPROVED** [review]

**Comment by FranciscoFactorLibre** [comment]
Hi @rousseldenis or @carolinafernandez-tecnativa 

Could you please check this PR?

Thanks

**Review by carolinafernandez-tecnativa — APPROVED** [review]
LGTM :) Thanks!!

**Review by andreampiovesana — APPROVED** [review]
merge?

File: stock_picking_mass_action/models/stock_picking.py [hunk 1]
```diff
@@ -10,9 +10,13 @@ class StockPicking(Model):
     _inherit = "stock.picking"
 
     @api.model
-    def check_assign_all(self):
+    def check_assign_all(self, other_domain=None):
```
Comment (rousseldenis) [inline stock_picking_mass_action/models/stock_picking.py]: I would have said `domain` without 'other' as this will be the only one at the end

Comment (RodrigoBM) [reply stock_picking_mass_action/models/stock_picking.py]: Done

**Review by rousseldenis — APPROVED** [review]

**Comment by RodrigoBM** [comment]
can @rousseldenis merge it?

**Pull request merged**